### DISCUSSION TITLE
Include Guards in Jinja

### DISCUSSION THREAD

**Discussion by LeStahL** [pallets/jinja#1988] [Q&A] [answered]
Hey, quick question - I noticed that `set` and `include` do not work together in a way that allows the "include guard" pattern:
```jinja
{%- if FBM_FRAG_J2 is not defined %}
{%- set FBM_FRAG_J2 = true %}
{# Guarded code goes here #}
{%- endif %}
```

I don't really have a good idea how to avoid duplicate (or recursive) includes without it - But I assume it must be possible somehow. What is the viable / intended strategy in jinja to achieve this?

**Comment by davidism** [comment, marked as answer]
Look up "namespace" in the docs.

**Reply by LeStahL** [reply]
thanks :)
